### DISCUSSION TITLE
How to apply default styles from a chakra component to a external component?

### DISCUSSION THREAD

**Discussion by csandman** [chakra-ui/chakra-ui#3651] [Q&A] [answered]
I am working on making a customized version of [react-select](https://react-select.com/home) to match it's styles to Chakra. There are actually a lot of direct equivalent's in Chakra to the pieces that make up react-select and react-select allows you to use your own components to make up the pieces.  However, for components such as MenuItem and other that make up the Menu component, they cannot be used without implementing them in full, in a directly nested manner.

So my question is, how can I access the default styles on a built in chakra component to apply to an external component.  It doesn't even have to be applied to an external component really, I just need to get the styles for a MenuItem (and a few others) without having the restrictions on the implementation of a component.

All I'd really need to do is something like:

```js
<Box as={MenuItem}></Box>
```

except without my app breaking because of it haha.  Any help is appreciated!

**Comment by csandman** [comment]
Side note, I'm only doing this because I need a multi-select field, so I'm definitely looking forward to the official chakra implementation of this component 😃

**Comment by csandman** [comment, marked as answer]
I ended up figuring it out, I used the `useMultiStyleConfig` hook: https://chakra-ui.com/docs/theming/component-style#consuming-multipart-style-config

And if anyone is curious, here is my wrapper component for react-select which styles everything in a way that matches Chakra UI: https://gist.github.com/csandman/c687a9fb4275112f281ab9a5701457e4

It is still a work in progress, but feel free to use it if you need a component like this!

**Comment by Abusayid693** [comment]
You can do this with any chakra component

```ts
import { Form } from 'formik';

 <Box as={Form} >
      .........
 </Box>

```
